### DISCUSSION TITLE
How much can Synthea output be customized?

### DISCUSSION THREAD

**Discussion by ToInf1n1tyAndBeyond** [synthetichealth/synthea#1323] [Q&A]
I am working on a project and we have a need for a large patient dataset with some very specific configurations. For each patient the base configuration appears to create a bundle with a single patient resource which is then followed by repeated groupings of Encounter, Condition, Claim, and ExplanationOfBenefit resources. The project focus is pharmacy providers and so we have specific interest in generating medication related patient samples.

My questions are...

1. Is it possible to simulate multiple medications for a single patient in either MedicationDispense or MedicationAdministration resources, contained in the Encounter resource?
2. Can the Condition, Claim, and ExplanationOfBenefit resources be suppressed in the output?
3. Can multiple patients be output in a single FHIR JSON file?

**Comment by jawalonoski** [comment]
> 1. Is it possible to simulate multiple medications for a single patient in either MedicationDispense or MedicationAdministration resources, contained in the Encounter resource?

Yes, a patient can have multiple Medication resources in a single Encounter.

No, a Medication resource cannot represent multiple medications. For instance,
  - _"The MedicationRequest resource allows requesting only a single medication. If a workflow requires requesting multiple items simultaneously, this is done using multiple instances of this resource."_ -- https://hl7.org/fhir/medicationrequest.html#scope

Synthea does not support MedicationDispense, but I'm sure you could post process MedicationAdministration or MedicationRequest to generate those as appropriate.

> 2. Can the Condition, Claim, and ExplanationOfBenefit resources be suppressed in the output?

Yes. You can set them in the `synthea.properties` file:

https://github.com/synthetichealth/synthea/blob/188ce6373ef9af428975c2f3e9631cac852df41f/src/main/resources/synthea.properties#L25-L30


> 3. Can multiple patients be output in a single FHIR JSON file?

No.

**Reply by ToInf1n1tyAndBeyond** [reply]
Thanks Jason. This is great. I was using the synthea.properties file from the Canadian filestore and it looks like the main .properties file has been updated with more options. A few follow up questions...

1. I added MedicationAdministration as a resource to be included...the Condition, Claim, and ExplanationOfBenefit resources are now suppressed but I don't see a MedicationAdministration resource in the Encounter. Am I missing something?

exporter.fhir.included_resources = MedicationAdministration

2. How do I limit the # of encounters per patient?

**Reply by jawalonoski** [reply]
> 1. I added MedicationAdministration as a resource to be included...the Condition, Claim, and ExplanationOfBenefit resources are now suppressed but I don't see a MedicationAdministration resource in the Encounter. Am I missing something?

You can only set included or excluded, but not both.

```properties
# only one of these may be set at a time, if both are set then both will be ignored. 
```

MedicationAdministrations will only appear if they are in Encounter. For example, in the Module Builder, while editing a `MedicationOrder` state:

![image](https://github.com/synthetichealth/synthea/assets/1054765/46a0cb22-9ac9-4a1a-8b9f-a041b4b3446f)


> exporter.fhir.included_resources = MedicationAdministration
> 
> 2. How do I limit the # of encounters per patient?

You don't. Either the module does that, or you can alter the years of history exported.

https://github.com/synthetichealth/synthea/blob/188ce6373ef9af428975c2f3e9631cac852df41f/src/main/resources/synthea.properties#L8-L10

**Reply by ToInf1n1tyAndBeyond** [reply]
I think I've updated the synthea.properties files as necessary (see screenshot). I see you have a screenshot from a GUI interface as well...do I need to set MedicationAdministration resource somewhere else? When I leave both the included and excluded parameters blank I get all resources (including MedicationRequest) and when I add MedicationAdministration to the included parameter, most of the resources are suppressed but I still get no MedicationAdministration resources. So the parameter file is being read and processed but I'm wondering if there's somewhere else I need to be adding drugs/meds...?

![image](https://github.com/synthetichealth/synthea/assets/137439218/eab9348f-68a4-4dfb-b87e-4f2ca89dac51)

**Reply by jawalonoski** [reply]
>  I see you have a screenshot from a GUI interface as well...do I need to set MedicationAdministration resource somewhere else? 

Not necessarily, the point is that in the modules some medications are prescribed and some are administered. The amount administered is not as frequent as those prescribed. If you want to change that, you have to edit the modules.

I just randomly generated 118 patients using the default settings and these are the FHIR R4 resources that were generated, including 2931 `MedicationAdministrations`.

| Resource | Count |
|---------|------:|
Observation | 73793
Claim | 22579
ExplanationOfBenefit | 22579
DiagnosticReport | 17752
Procedure | 13975
MedicationRequest | 12394
DocumentReference | 10185
Encounter | 10185
Condition | 6095
Medication | 2931
MedicationAdministration | 2931
Immunization | 1755
SupplyDelivery | 1363
CarePlan | 435
CareTeam | 435
Location | 289
Organization | 288
Practitioner | 288
PractitionerRole | 288
Device | 162
ImagingStudy | 140
Patient | 118
Provenance | 118
AllergyIntolerance | 81

**Reply by ToInf1n1tyAndBeyond** [reply]
How do I edit the frequency of the med requests and med admins?

Could I ask you to try running a set of patients with exporter.fhir.included_resources = MedicationAdministration in your Synthea.properties file (and excluded_resources is not set) and tell me if you get MedicationAdministration resources? When I do this I get Encounter, Location, Organization, Patient, Practitioner and PractitionerRole resources as output but no MedicationAdministration resources.

I do get MedicationAdministration resources when both the included_resources and excluded_resources params are left blank but we're trying to generate patients with only the resources that we need so conserve file size and disk space.

**Reply by jawalonoski** [reply]
> How do I edit the frequency of the med requests and med admins?

You have to edit the modules using the module builder. https://synthetichealth.github.io/module-builder/

You can manually add MedicationOrder states, or edit existing ones to make them chronic prescriptions, add details on dosage and frequency, or make them administrations.

> Could I ask you to try running a set of patients with exporter.fhir.included_resources = MedicationAdministration in your Synthea.properties file (and excluded_resources is not set) and tell me if you get MedicationAdministration resources? When I do this I get Encounter, Location, Organization, Patient, Practitioner and PractitionerRole resources as output but no MedicationAdministration resources.
> 
> I do get MedicationAdministration resources when both the included_resources and excluded_resources params are left blank but we're trying to generate patients with only the resources that we need so conserve file size and disk space.

You have to include all `Medication*` resources because they are linked together.

```properties
exporter.fhir.included_resources = Medication,MedicationRequest,MedicationAdministration
```

**Comment by ToInf1n1tyAndBeyond** [comment]
The medications worked. Thanks.

If I have files already downloaded on my local system, what file do I open/select to use the module builder?

**Reply by jawalonoski** [reply]
The module builder is a web app, it is not distributed as part of Synthea itself.

If you really want to run a local copy, you can get it from https://github.com/synthetichealth/module-builder

However, unless you are disconnected from the internet, there is no reason to do that.

Go to https://synthetichealth.github.io/module-builder/, and either "Drop files here or click to open saved modules" or click "Paste JSON" and copy and paste. That is how you edit your local files. When you are done editing, you click the "Download" link at the top of the page and save them back into your distribution.
